rclrs_examples/CMakeLists.txt

```diff
@@ -45,8 +45,8 @@ add_custom_target(copy_crate ALL
 
 add_custom_command(
     TARGET copy_crate
-    COMMAND ${CMAKE_COMMAND} -E remove_directory "${CMAKE_BINARY_DIR}/src"
-    COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_SOURCE_DIR}/src" "${CMAKE_BINARY_DIR}/src"
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_SOURCE_DIR}/src/rclrs_publisher.rs" "${CMAKE_BINARY_DIR}/src/rclrs_publisher.rs"
```

Comment:
This will spawn a new process, whereas `configure_file` runs within the same CMake process. Moreover, with `configure_file` other targets can depend directly on the files, so you don't need a custom command. In general, `add_custom_command` should be used as the last resort and use the functions and macros provided by CMake.

Reply:
Sorry, a comment was coming along with the update but I was delayed.
I tried to make it work with configure_file but it doesn't seem to trigger on file change so I thought that using copy_if_different would be an acceptable workaround for now.